rules/store.go

```diff
@@ -0,0 +1,104 @@
+package rules
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+// AlertStore provides persistent storage of alert state.
+type AlertStore interface {
+	// SetAlerts stores the provided list of alerts for a rule.
+	SetAlerts(key uint64, alerts []*Alert) error
+	// GetAlerts returns a list of alerts for each alerting rule,
+	// alerting rule is identified by a fingerprint of its config.
+	GetAlerts(key uint64) (map[uint64]*Alert, error)
+}
+
+// FileStore implements the AlertStore interface.
+type FileStore struct {
+	logger       log.Logger
+	alertsByRule map[uint64][]*Alert
+	//protects the `alertsByRule` map
+	stateMtx sync.RWMutex
+	path     string
+}
+
+func NewFileStore(l log.Logger, storagePath string) *FileStore {
+	s := &FileStore{
+		logger:       l,
+		alertsByRule: make(map[uint64][]*Alert),
+		path:         storagePath,
+	}
+	s.initState()
+	return s
+}
+
+// initState reads the state from file storage into the alertsByRule map
+func (s *FileStore) initState() {
+	file, err := os.OpenFile(s.path, os.O_RDWR|os.O_CREATE, 0o666)
+	if err != nil {
+		level.Error(s.logger).Log("msg", "Failed reading alerts state from file", "err", err)
+		return
+	}
+	defer file.Close()
+
+	var alertsByRule map[uint64][]*Alert
+	err = json.NewDecoder(file).Decode(&alertsByRule)
+	if err != nil {
+		level.Error(s.logger).Log("msg", "Failed reading alerts state from file", "err", err)
+		return
```

Comment:
Why do we need to return early here? I don't see why we cannot continue to the end of the function